src/js/program/media-controller.js

```diff
@@ -138,6 +138,9 @@ export default class MediaController extends Eventable {
                 // Start firing visualQuality once playback has started
                 mediaModel.off(MEDIA_VISUAL_QUALITY, null, this);
                 mediaModel.change(MEDIA_VISUAL_QUALITY, (changedMediaModel, eventData) => {
+                    if (!eventData) {
+                        return;
```

Comment:
Providers should not dispatch visualQuality events without data. The guard should be in the provider before "visualQuality" is set on mediaModel.